src/main/js/text_track/text_track_helper.js

```diff
@@ -0,0 +1,151 @@
+   * @public
+   * @param {Object} trackData An object with the relevant properties to set on the text track object.
+   */
+  addTrack(trackData = {}) {
+    if (!this.video) {
+      return;
+    }
+    const track = document.createElement('track');
+    track.id = trackData.id;
+    track.kind = trackData.kind;
+    track.label = trackData.label;
+    track.srclang = trackData.srclang;
+    track.src = trackData.src;
+    this.video.appendChild(track);
+  }
+
+  /**
+   * Finds a text track by id and sets its label to the given value. This operation
+   * is only possible for tracks that were manually added by the plugin.
+   * @public
+   * @param {String} trackId The dom id of the text track to update
+   * @param {String} label The new label to be set on the text track
+   */
+  updateTrackLabel(trackId, label = '') {
+    if (this.video && trackId) {
+      const trackElement = this.video.querySelector(`#${trackId}`);
+
+      if (trackElement) {
+        trackElement.setAttribute('label', label);
+      }
+    }
+  }
+
+  /**
+   * Allows executing Array.prototype.forEach on the video's TextTrackList.
+   * @public
+   * @param {Function} callback A function to execute for existing text track.
+   */
+  forEach(callback) {
+    if (!this.video || !this.video.textTracks) {
+      return;
+    }
+    Array.prototype.forEach.call(this.video.textTracks, callback);
+  }
+
+  /**
+   * Allows executing Array.prototype.filter on the video's TextTrackList.
+   * @public
+   * @param {Function} callback A predicate function to test each element of the array.
+   * @return {Array} An array with all the TextTrack objects that match the filter criteria.
+   */
+  filter(callback) {
+    if (!this.video || !this.video.textTracks) {
+      return [];
+    }
+    return Array.prototype.filter.call(this.video.textTracks, callback);
+  }
+
+  /**
+   * Allows executing Array.prototype.find on the video's TextTrackList.
+   * @public
+   * @param {Function} callback A function to execute on each value in the array.
+   * @return {TextTrack} The first TextTrack object that matches the search criteria or undefined if there are no matches.
+   */
+  find(callback) {
+    if (!this.video || !this.video.textTracks) {
+      return;
+    }
+    let track = Array.prototype.find.call(this.video.textTracks, callback);
+    return track;
+  }
+
+  /**
+   * Finds the TextTrack object that matches a key, which can be either the language
+   * code of the track or a track id associated with the track on a TextTrackMap.
+   * @public
+   * @param {String} languageOrId The language or track id of the track we want to find.
+   * @param {TextTrackMap} textTrackMap A TextTrackMap that contains metadata for all of the video's TextTrack objects.
+   * @return {TextTrack} The first TextTrack object that matches the given key or undefined if there are no matches.
+   */
+  findTrackByKey(languageOrId, textTrackMap = new TextTrackMap()) {
```

Comment:
Is it possible for two text tracks to have the same exact language key? If so, should this return all the tracks that have that key? Also, who is using this function. I tried searching this PR for ```findTrackByKey``` and coulnd't find a usage.

Reply:
Yes, good catch. I will have to tweak this in order to account for that scenario. This is called by `executeSetClosedCaptions` btw